services/graphService.js

```diff
@@ -58,28 +59,46 @@ GraphService.prototype.publicRankedList = function() {
 	return dfd.promise
 }
 
-GraphService.prototype.userGraphs = function(username) {
+GraphService.prototype.userGraphs = function(username, filter, offset, limit) {
```

Comment:
the problem with allowing Controller to set filters, is that we can't control the indexing in the model/"service" anymore.